packages/abap-deploy-config-sub-generator/src/app/index.ts

```diff
@@ -0,0 +1,316 @@
+            this.answers.transport =
+                getTransportAnswer(this.options as AbapDeployConfigAnswersInternal) || getTransportAnswer(this.answers);
+        }
+    }
+
+    /**
+     * Reconcile answers with options.
+     *
+     * Options may be passed from parent generator, or from the command line.
+     */
+    private _reconcileAnswersWithOptions(): void {
+        this._processAbapTargetAnswers();
+        this._processBspAppAnswers();
+        this.answers.index = this.options.index || this.answers.index;
+        this.answers.description = this.options.description || this.answers.description;
+        this.answers.overwrite = this.options.overwrite || this.answers.overwrite;
+    }
+
+    public async writing(): Promise<void> {
+        if (!this.launchDeployConfigAsSubGenerator) {
+            await this._writing();
+        }
+    }
+
+    private async _writing(): Promise<void> {
+        if (this.abort || this.answers.overwrite === false) {
+            return;
+        }
+        await generateAbapDeployConfig(
+            this.destinationPath(),
+            {
+                target: {
+                    url: this.answers.url,
+                    client: this.answers.client,
+                    scp: this.answers.scp,
+                    destination: this.answers.destination,
+                    authenticationType: this.answers.isS4HC ? AuthenticationType.ReentranceTicket : undefined // only reentrance ticket is relevant for writing to deploy config
+                },
+                app: {
+                    name: this.answers.ui5AbapRepo,
+                    description: this.answers.description,
+                    package: this.answers.package,
+                    transport: this.answers.transport
+                },
+                index: this.answers.index
+            } as AbapDeployConfig,
+            {
+                baseFile: this.options.base,
+                deployFile: this.options.config
+            },
+            this.fs
+        );
+    }
+
+    public install(): void {
+        if (!this.launchDeployConfigAsSubGenerator && this.answers.overwrite !== false) {
+            this._install();
+        }
+    }
+
+    private _install(): void {
+        if (!this.options.skipInstall) {
+            this.spawnCommand('npm', [
+                'install',
+                '--no-audit',
+                '--no-fund',
+                '--silent',
+                '--prefer-offline',
+                '--no-progress'
+            ]);
+        } else {
+            DeploymentGenerator.logger?.info(t('info.skippedInstallation'));
+        }
+    }
+
+    public async end(): Promise<void> {
+        if (this.abort || this.answers.overwrite === false) {
+            return;
+        }
+
+        // Delayed process of deploy configuration generation if integrated with app generator
+        if (this.launchDeployConfigAsSubGenerator) {
+            await this._initializing();
```

Comment:
Not sure why we are initing and writing in the end phase, writing should be done in memory during the writing phase only.